packages/source-iotsitewise/src/time-series-data/subscribeToTimeSeriesData.ts

```diff
@@ -1,76 +1,70 @@
-import { DescribeAssetModelResponse } from '@aws-sdk/client-iotsitewise';
-import { completeDataStreams } from '../completeDataStreams';
+import { DataModule, DataModuleSubscription, TimeSeriesData, SubscriptionUpdate } from '@iot-app-kit/core';
 import { SiteWiseDataStreamQuery } from './types';
-import { MinimalViewPortConfig } from '@synchro-charts/core';
-import {
-  ErrorDetails,
-  DataModule,
-  DataModuleSubscription,
-  DataStream,
-  TimeSeriesData,
-  SubscriptionUpdate,
-} from '@iot-app-kit/core';
 import { SiteWiseAssetSession } from '../asset-modules';
+import { SiteWiseAlarmModule } from '../alarms/iotevents';
+import { fetchAssetModelsFromQuery } from '../asset-modules/util/fetchAssetModelsFromQuery';
+import { fetchAlarmsFromQuery } from '../alarms/iotevents/util/fetchAlarmsFromQuery';
+import { CreateTimeSeriesDataStore } from './store';
+
+const initialState = {
+  dataStreams: [],
+  annotations: {},
+  assetModels: {},
+  alarms: {},
+  errors: {},
+};
 
 export const subscribeToTimeSeriesData =
-  (dataModule: DataModule, assetModuleSession: SiteWiseAssetSession) =>
+  (dataModule: DataModule, assetModuleSession: SiteWiseAssetSession, alarmModule: SiteWiseAlarmModule) =>
   ({ queries, request }: DataModuleSubscription<SiteWiseDataStreamQuery>, callback: (data: TimeSeriesData) => void) => {
-    let dataStreams: DataStream[] = [];
+    const store = new CreateTimeSeriesDataStore({ initialState, callback });
 
-    let viewport: MinimalViewPortConfig;
+    const { update, unsubscribe } = dataModule.subscribeToDataStreams({ queries, request }, (data) => {
+      store.setState({
+        dataStreams: data.dataStreams,
+        viewport: data.viewport,
+      });
+    });
 
-    const assetModels: Record<string, DescribeAssetModelResponse> = {};
+    const updateAssetModels = (queries: SiteWiseDataStreamQuery[]) => {
+      (async () => {
+        for await (const response of fetchAssetModelsFromQuery({ queries, assetModuleSession })) {
+          const assetModels = 'assetModels' in response && response.assetModels;
+          const errors = 'errors' in response && response.errors;
 
-    const errors: Record<string, ErrorDetails> = {};
+          if (assetModels) {
+            store.setState({ assetModels });
+          }
 
-    const emit = () => {
-      callback({
-        dataStreams: completeDataStreams({ dataStreams, assetModels }),
-        viewport,
-      });
+          if (errors) {
+            store.setState({ errors });
```

Comment:
if there are not errors would we want to set it to undefined?

Reply:
Yes, @boweihan brought this up as well. We will have to revisit this and improve error handling for asset models.